app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -133,6 +137,14 @@ public boolean onLongClick(View v) {
         });
     }
 
+    private void validateYonaPasswordEncryption(){
+        if(!sharedUserPreferences.getBoolean(YONA_ENCRYPTION_UPGRADED,false)){
+            YonaApplication.getEventChangeManager().getSharedPreference().upgradeYonaPasswordEncryption();
+        }else {
+            sharedUserPreferences.edit().putBoolean(YONA_ENCRYPTION_UPGRADED,true);
```

Comment:
In the else-branch, the flag is already true. Instead, add this to the end of the if-branch:
```
sharedUserPreferences.edit().putInt(YONA_ENCRYPTION_METHOD, EncryptionMethod.ENHANCED_STILL_BASED_ON_SERIAL);
```

Reply:
For old versions of applications this value will be null so setting it to true to avoid the upgrade on second launch

Reply:
I have updated the code now with Enum.